ska_sun/tests/test_sun.py

```diff
@@ -35,16 +39,46 @@
 )
 
 
-@pytest.mark.parametrize("pitch, rolldev", exp_pitch_rolldev)
-def test_allowed_rolldev(pitch, rolldev):
-    # Test array of pitchs and allowed roll dev
-    assert np.isclose(allowed_rolldev(pitch), rolldev)
+@contextlib.contextmanager
+def clear_roll_table_lazy_val():
+    """Context manager to ensure the ROLL_TABLE LazyVal global is cleared before and
+    after the test.
 
+    This is needed to change the chandra_models version. Kinda yucky but there is no
+    convenient way to do this otherwise."""
+    if hasattr(ska_sun.sun.ROLL_TABLE, "_val"):
+        ska_sun.sun.ROLL_TABLE = LazyVal(ska_sun.sun.load_roll_table)
+    yield
+    if hasattr(ska_sun.sun.ROLL_TABLE, "_val"):
+        ska_sun.sun.ROLL_TABLE = LazyVal(ska_sun.sun.load_roll_table)
 
-def test_allowed_rolldev_vector():
-    assert np.allclose(
-        allowed_rolldev(exp_pitch_rolldev[:, 0]), exp_pitch_rolldev[:, 1]
-    )
+
+@pytest.mark.parametrize("pitch, rolldev", exp_pitch_rolldev)
+def test_allowed_rolldev(pitch, rolldev, monkeypatch):
+    monkeypatch.setenv("CHANDRA_MODELS_DEFAULT_VERSION", "3.49")
+    # Test array of pitchs and allowed roll dev
+    with clear_roll_table_lazy_val():
+        assert np.isclose(allowed_rolldev(pitch), rolldev)
+
+
+def test_allowed_rolldev_vector(monkeypatch):
+    # Force reload of roll table
+    monkeypatch.setenv("CHANDRA_MODELS_DEFAULT_VERSION", "3.49")
+    with clear_roll_table_lazy_val():
+        assert np.allclose(
+            allowed_rolldev(exp_pitch_rolldev[:, 0]), exp_pitch_rolldev[:, 1]
+        )
+
+
+def test_duplicate_pitch_rolldev(monkeypatch):
+    # This is a commit of the 2023_020 pitch/roll constraint file that is exactly what
+    # was provided by the FOT (except adding the header columns). It contains several
+    # duplicate pitch values, including pitch=85.5 with 12.436 and 17.5 roll dev vals.
+    # The test is to make sure that the code handles this.
+    monkeypatch.setenv("CHANDRA_MODELS_DEFAULT_VERSION", "68670fc")
```

Comment:
Regarding adding a unit test using a specific version of the file, does the clear_roll_table_lazy_val fix suggest that if we want to add tests using different versions of the chandra_models pitch/roll table in other packages we'll need to use that code there too?  I'm wondering if we should use the ska_helpers.chandra_models info with the LazyVal to handle this.

Reply:
I'm going to just close this comment, but keep it in mind as an issue for myself (ska_sun uses a cache for the roll table and, unlike the other models, will not really respect CHANDRA_MODELS_DEFAULT_VERSION switches unless the cache is cleared).

Reply:
And who knows, maybe we should just make that change now. :smile:

Reply:
I'll take a crack at it. 5 minutes? Famous last words.